var/spack/repos/builtin/packages/casacore/package.py

```diff
@@ -53,8 +53,8 @@ class Casacore(CMakePackage):
     depends_on("lapack")
     depends_on("cfitsio")
     depends_on("wcslib@4.20:+cfitsio")
-    depends_on("fftw@3.0.0: precision=float,double", when="@3.4.0:")
-    depends_on("fftw@3.0.0: precision=float,double", when="~fftpack")
+    depends_on("fftw-api@3 precision=float,double", when="@3.4.0:")
+    depends_on("fftw-api@3 precision=float,double", when="~fftpack")
```

Comment:
This is wrong, in the sense that virtual dependencies do not have variant attached at the moment. Can you add back the additional constraint on `fftw`, if needed, and clean the one on the virtual dependency?

Reply:
Hi @alalazo, is this documented somewhere? I will say that this worked. And to be clear, you are saying have something like this?
```python
    depends_on("fftw-api@3", when="@3.4.0:")
    depends_on("fftw-api@3", when="~fftpack")
    depends_on("fftw precision=float,double", when="~fftpack")
```

Reply:
@pelahi To be fair, it is badly documented at https://spack.readthedocs.io/en/latest/packaging_guide.html#virtual-dependencies but we usually stress this part in tutorials. 

>  I will say that this worked.

I'm pretty sure this was just interpreted as:
```python
depends_on("fftw-api@3", when="@3.4.0:")
depends_on("fftw-api@3", when="~fftpack")
```

> And to be clear, you are saying have something like this?

Almost. This:
```python
depends_on("fftw-api@3", when="@3.4.0:")
depends_on("fftw-api@3", when="~fftpack")
depends_on("fftw precision=float,double", when="^fftw")
```
should work and allow you to pick any other `fftw-api` provider even if `~fftpack` (assuming that makes sense).